Added configuration for Github API scopes.

jason-dour opened the pull request:
- Added configuration contribution.
- Added default scopes read in from configuration.

Addresses #9

jason-dour committed:
Added configuration for Github API scopes.

- Added configuration contribution.
- Added default scopes read in from configuration.

jason-dour commented:
I'll admit this is my first foray into VSCode extension development, so I can only say I **think** this PR addresses #9 as you described.  It does appear to have proper scope configuration.  Once I added my custom scopes, it was able to execute my query that was normally disallowed by the defaults of `["repo","workflow"]`.

joyceerhl commented on package.json:
```diff
@@ -18,29 +18,42 @@
+				"github-graphql-nb.scopes": {
+					"type": "array",
+					"default": [ "repo", "workflow" ],
+					"markdownDescription": "The authentication scopes to request when signing into GitHub. See [Available Scopes](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes) in Github Documentation for more information.",
```
Nice description :clap: I appreciate the link to the documentation! 

```suggestion
					"markdownDescription": "The authentication scopes to request when signing into GitHub. See [Available Scopes](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes) in GitHub Documentation for more information.",
```

joyceerhl commented on package-lock.json:
```diff
@@ -1,4049 +1,8 @@
-	"version": "0.0.8",
-	"lockfileVersion": 2,
+	"version": "0.1.0",
+	"lockfileVersion": 1,
```
Ah I forgot to bump the extension version in the lockfile. Still, the lockfile probably doesn't need to change this much (VS Code develops against node 16.x, which comes with npm 7, and which generates v2 lockfiles). Would you mind either updating npm and rerunning `npm i`, or dropping the change to the package-lock.json? Thanks a ton!

jason-dour replied:
If I update `cross-fetch` and `node-fetch` I'm still seeing vuln warnings on mocha, but being a breaking change.  Leaving that one alone.

jason-dour replied:
Fixed this as best as I understand it.

joyceerhl commented on src/extension.ts:
```diff
@@ -11,7 +11,7 @@ export function activate(context: vscode.ExtensionContext) {
 
 
-const authorizationScopes = ['repo', 'workflow'];
+const authorizationScopes = vscode.workspace.getConfiguration('github-graphql-nb').get('scopes') as string[];
```
If we read the value of this setting just once, a user would have to reload the VS Code window for the new auth scopes to be picked up. A small suggestion here: in the constructor of the `OctokitController` after line 25 in this file, we could use the `vscode.workspace.onDidChangeConfiguration` event to detect a change to this setting's value, and update the authorization scopes we ask for and clear out the existing auth session:

```ts
vscode.workspace.onDidChangeConfiguration((e) => {
    if (e.affectsConfiguration('github-graphql-nb')) {
        authorizationScopes = vscode.workspace.getConfiguration('github-graphql-nb').get('scopes') ?? []; // Update our cached auth scopes
        this.clearAuthenticationSession(); // Clear out the existing auth session if requested scopes changed
    }
});
```

That way, after you change the value of this setting, you can simply click run and the extension will ask you to reauth with the new scopes before executing your query. This is probably slightly more user friendly than requiring the user to do a full reload. What do you think?

jason-dour replied:
I wondered about that. I'll take a look at confit change. Thanks. 😄

jason-dour replied:
Added onDigChangeConfiguration.

jason-dour committed:
Update package.json

Co-authored-by: Joyce Er <joyceerhl@gmail.com>

jason-dour committed:
Update package-lock.json

jason-dour committed:
Update extension.ts

jason-dour commented:
Made the recommended changes.  I think it addresses all the areas you mentioned.

My testing showed a failure of a query, config change with proper scope, and the query could then execute cleanly.

Let me know how else it can be improved.

jason-dour committed:
Make scopes editable in UI.

Declared items type to allow for editable lists in the UI for the configuration scopes.

jason-dour commented:
Added one last change.  Made the configuration editable in the UI by declaring the item types for the array.  This should make it more accessible and easy to edit, while still allowing for copy pasta editing when desired.

joyceerhl reviewed (approved):
:clap: thank you so much, this is awesome!
